service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java

```diff
@@ -613,6 +609,25 @@ public PaymentSystemResponse saveFullOrderToDB(OrderResponseDto dto, String uuid
         return processPayment(dto, order, sumToPayInCoins, currentUser);
     }
 
+    @Async
+    public void checkIfOrderIsNotPayedAndSendEmailAsync(Order order, Long sumToPayInCoins) {
+        try {
+            Thread.sleep(1000);
```

Comment:
Please replace the Thread.sleep() with the Spring TaskScheduler. The sleep() does not align with the asynchronous execution, blocking the thread and making it do nothing. TaskScheduler, on the other hand, schedules the task in one second, making the thread available for other tasks

Reply:
I'll deal with it